### PR TITLE
Rewrite to use WebSocketListener instead of sta-websocket-sharp

### PR DESCRIPTION
**Pull request by trinitrot0luene**
- [x] Re-implement required session infrastructure
- [x] Add support for the client connection & auth flow
~~- [ ] gRPC listener (MagicOnion) for receiving events~~
~~- [ ] Pipeline for event distribution & validation~~

### CODE REVIEW

File: src/SessionServer.API.Entities/WebSocket/Ident/IdHost.cs [hunk 1]
```diff
@@ -1,16 +1,43 @@
...
+
+using System;
 using System.Collections.Generic;
 using System.Runtime.Serialization;
 using System.Text;
 
 namespace SessionServer.API.Entities.WebSocket
 {
-    public struct IdHost
+    public class IdHost
```
Comment (Izumemori) [inline src/SessionServer.API.Entities/WebSocket/Ident/IdHost.cs]: Not a critique but why did you change it to a class? Doesn't seem necessary

Comment (trinitrot0luene) [reply src/SessionServer.API.Entities/WebSocket/Ident/IdHost.cs]: Because it'd be copied when using the SendMessageAsync() helper. I could change it to accept a Func<T> or Payload<T> overload and keep it a struct, or use classes everywhere instead.
